ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -138,6 +138,29 @@ impl TraceBuilder {
         Ok(())
     }
 
+    /// Process promotions inside an otherwise outlined block.
+    fn process_promotions_only(&mut self, bid: &aot_ir::BBlockId) -> Result<(), CompilationError> {
+        let blk = self.aot_mod.bblock(bid);
+
+        for inst in blk.insts.iter() {
+            match inst {
+                aot_ir::Inst::Promote { val, tyidx, .. } => match val {
+                    aot_ir::Operand::LocalVariable(_) => {
+                        let width_bits = match self.aot_mod.type_(*tyidx) {
+                            aot_ir::Ty::Integer(it) => it.num_bits(),
+                            _ => unreachable!(),
+                        };
+                        let width_bytes = usize::try_from(width_bits).unwrap() / 8;
```

Comment:
Oopsy. An `i1` would report a byte size of 0! I know we can't promote an `i1`, but still.

See this [jit ir routine](https://github.com/ykjit/yk/blob/3179eb557cb41201ba3b8a3ed12b9afc48fe5c5d/ykrt/src/compile/jitc_yk/jit_ir/mod.rs#L1001) for inspiration.

Every time we are tempted to divide a bit size by 8, an alarm bell should ring :)

Reply:
I just copied this from `handle_operand`. We should fix it there too, but I'm not sure this PR is the right place.

Reply:
OK, please follow this PR up with the fix. It'll surely bite us later (again).